contracts/utils/Counters.sol

```diff
@@ -23,10 +23,16 @@ library Counters {
     }
 
     function increment(Counter storage counter) internal {
-        counter._value += 1;
+        unchecked {
+            counter._value += 1;
+        }
     }
 
     function decrement(Counter storage counter) internal {
-        counter._value = counter._value - 1;
+        uint256 value = counter._value;
+        require(value > 0, "Counter: decrement underflow");
```

Comment:
Sorry, one last comment I didn't notice before. I don't know if this is the revert reason we had before but this should be "overflow". Underflow is for floating point numbers only.

Reply:
The previous error was the safemath one. I believed it would be better to have a more explicit one.

Do you want `Counter: decrement overflow` or would you prefer something different ?

Reply:
"decrement overflow" sounds good to me.

Reply:
without the `Counter:` prefix ?

Reply:
No, sorry, `Counter: decrement overflow`.